frontend/packages/console-app/src/redux/reducers/quick-start-reducer.ts

```diff
@@ -1,31 +1,142 @@
+import { Map, fromJS } from 'immutable';
 import { QuickStartSidebarActions, Actions } from '../actions/quick-start-actions';
 import { RootState } from '@console/internal/redux';
+import {
+  AllQuickStartStates,
+  QuickStartState,
+  QuickStartStatus,
+  QuickStartTaskStatus,
+} from '../../components/quick-starts/utils/quick-start-types';
 
 export const quickStartReducerName = 'quickStart';
 
-type State = {
-  isExpanded: boolean;
-  activeQuickStartID: string;
-};
+export const QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY = 'bridge/quick-start-redux-state';
+
+type State = Map<string, string | Map<string, string | number>>;
 
-const initialState: State = {
-  isExpanded: false,
-  activeQuickStartID: '',
+const getDefaultQuickStartState = (totalTasks: number) => {
+  const defaultQuickStartState = {
+    status: QuickStartStatus.NOT_STARTED,
+    taskNumber: -1,
+  };
+  if (totalTasks) {
+    for (let i = 0; i < totalTasks; i++) {
+      defaultQuickStartState[`taskStatus${i}`] = QuickStartTaskStatus.INIT;
+    }
+  }
+  return Map(defaultQuickStartState);
 };
 
-export default (state = initialState, action: QuickStartSidebarActions) => {
-  if (action.type === Actions.SetActiveQuickStart) {
-    return {
-      isExpanded: !!action.payload.activeQuickStartID,
-      activeQuickStartID: action.payload.activeQuickStartID || state.activeQuickStartID,
+export const getInitialState = (): State => {
+  const persistedState = localStorage.getItem(QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY);
+
+  let initialState;
+  if (persistedState) {
+    initialState = JSON.parse(persistedState);
+  } else {
+    initialState = {
+      activeQuickStartId: '',
     };
+    localStorage.setItem(QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY, JSON.stringify(initialState));
   }
 
-  return state;
+  return fromJS(initialState);
 };
 
-export const isQuickStartDrawerExpanded = (state: RootState): boolean =>
-  !!state.plugins?.console?.[quickStartReducerName]?.isExpanded;
+const persistState = (state: State) =>
+  localStorage.setItem(QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY, JSON.stringify(state.toJS()));
+
+export default (state = getInitialState(), action: QuickStartSidebarActions) => {
+  switch (action.type) {
+    case Actions.SetActiveQuickStart: {
+      const { quickStartId, totalTasks } = action.payload;
+      const newState = state.withMutations((mutationState) =>
+        mutationState
+          .update('activeQuickStartId', (activeId) => {
+            return quickStartId !== activeId ? quickStartId : '';
+          })
+          .updateIn(['allQuickStartStates', quickStartId], (quickStartState) => {
+            if (!quickStartId || quickStartState) {
+              return quickStartState;
+            }
+
+            return getDefaultQuickStartState(totalTasks);
+          }),
+      );
+      persistState(newState);
```

Comment:
Reducers should be free of side effects, incl. storing data. A good option is the library [redux-persist](https://github.com/rt2zz/redux-persist). It's also possible to persist just single reducers, it is not required to persist everything.

I expect this takes some time, so I'm fine with this at the moment. WDYT? Should we create a ticket to change this later?

Reply:
Yeah, this is not the first reducer that we have written that had side-effects. I believe the Topology filters work like this too. I'll find the ticket I logged for that, perhaps we just expand to cover this scenario too.

Reply:
I actually looked at `redux-persist` for this but didn't use it considering it would e a lot of overhead to firrst setup everything to work with it. As it is the PR has become huge. We can look at introducing `redux-persist` later on.